biocypher_metta/adapters/hgnc_processor.py

```diff
@@ -2,93 +2,125 @@
+        # Check if we have a cached result from the last 30 minutes
+        if self.last_update_check and (current_time - self.last_update_check) < timedelta(minutes=30):
+            return self.last_check_result
+
+        self.last_update_check = current_time
+
         if not os.path.exists(self.version_file_path):
-            print(f"Version file not found. Current version: {self.current_version}")
+            print("HGNC data: Version file not found. Update needed.")
+            self.last_check_result = True
             return True
+    
         with open(self.version_file_path, 'r') as f:
-            stored_version = f.read().strip()
-        print(f"Stored version: {stored_version}")
-        print(f"Current version: {self.current_version}")
-        return self.current_version != stored_version
+            last_update_str = f.read().strip()
+    
+        try:
+            last_update = datetime.fromisoformat(last_update_str)
+            time_since_update = current_time - last_update
+            update_needed = time_since_update > self.update_interval
+        
+            if update_needed:
+                print(f"HGNC data: Last updated {time_since_update.days} days ago. Update needed.")
+            else:
+                print(f"HGNC data: Last updated {time_since_update.days} days ago. No update needed.")
+            
+            self.last_check_result = update_needed
+            return update_needed
+        except ValueError:
+            print("HGNC data: Invalid date format in version file. Forcing update.")
+            self.last_check_result = True
+            return True
 
-    def save_version(self):
-        """Save the current version to file"""
+    def save_update_time(self):
+        """Save the current time as the last update time"""
         os.makedirs(os.path.dirname(self.version_file_path), exist_ok=True)
+        current_time = datetime.now().isoformat()
         with open(self.version_file_path, 'w') as f:
-            f.write(self.current_version)
+            f.write(current_time)
+        print(f"HGNC data: Saved update time: {current_time}")
 
     def update_hgnc_data(self):
         """Update HGNC data if needed"""
-        if not self.check_version() and os.path.exists(self.pickle_file_path):
-            print("Using existing HGNC data.")
+        if not self.check_update_needed() and os.path.exists(self.pickle_file_path):
+            print("HGNC data: Using existing data.")
             self.load_data()
             return
 
-        print("Updating HGNC data...")
-        url = "https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/tsv/hgnc_complete_set.txt"
-        # response = requests.get(url)
-        # response.raise_for_status()
+        print("HGNC data: Updating...")
+        url = "https://www.genenames.org/cgi-bin/download/custom?col=gd_app_sym&col=gd_prev_sym&col=gd_aliases&col=gd_pub_ensembl_id&status=Approved&hgnc_dbtag=on&order_by=gd_app_sym_sort&format=text&submit=submit"
```

Comment:
please define this link in the constructor.

Reply:
I have pushed the new code with the changes.